### PR TITLE
Add a new entity Category,  Category repository, DTO, mapper, service…

### PR DESCRIPTION
**Pull request by AgroFix**
…, controller

### CODE REVIEW

File: src/main/java/mate/academy/bookstore/config/SecurityConfig.java [hunk 1]
```diff
@@ -35,8 +35,8 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
                 .csrf(AbstractHttpConfigurer::disable)
```
Comment (BolsunovaNataliia) [inline src/main/java/mate/academy/bookstore/config/SecurityConfig.java:35]: Good job! But I have some comments

File: src/main/java/mate/academy/bookstore/config/SecurityConfig.java [hunk 2]
```diff
@@ -35,8 +35,8 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
...
                 .authorizeHttpRequests(
                         auth -> auth
-                                .requestMatchers("/api/auth/**", "/error", "/swagger-ui/**"
-                                ,"/v3/api-docs/**")
+                                .requestMatchers("/api/auth/**", "/error",
+                                        "/swagger-ui/**","/v3/api-docs/**")
```
Comment (BolsunovaNataliia) [inline src/main/java/mate/academy/bookstore/config/SecurityConfig.java:39]: It must be carried out in application.property.

File: src/main/resources/db/changelog/changes/12-create-books-categories-table.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+databaseChangeLog:
+  - changeSet:
+      id: create-books-categories-table
```
Comment (BolsunovaNataliia) [inline src/main/resources/db/changelog/changes/12-create-books-categories-table.yaml]: May be create-books_categproes-table?

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 4]
```diff
@@ -17,3 +17,11 @@ databaseChangeLog:
       file: db/changelog/changes/08-insert-roles.yaml
   - include:
       file: db/changelog/changes/09-insert-users-roles.yaml
+  - include:
+      file: db/changelog/changes/10-create-categories-table.yaml
+  - include:
+      file: db/changelog/changes/11-insert-categories.yaml
+  - include:
+      file: db/changelog/changes/12-create-books-categories-table.yaml
```
Comment (BolsunovaNataliia) [inline src/main/resources/db/changelog/db.changelog-master.yaml]: May be 12-create-books_categories-table.yaml

**Review by BolsunovaNataliia — APPROVED** [review]

File: src/main/java/mate/academy/bookstore/controller/CategoryController.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+@RestController
+@RequiredArgsConstructor
+@RequestMapping(value = "/categories")
+@Tag(name = "Category management", description = "Endpoints for managing categories")
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final BookService bookService;
+
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping
```
Comment (d1sam) [inline src/main/java/mate/academy/bookstore/controller/CategoryController.java:36]:
```suggestion
    @ResponseStatus(HttpStatus.CREATED)
    @PostMapping
```

File: src/main/java/mate/academy/bookstore/controller/CategoryController.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    @Operation(summary = "Update category", description = "Update category by id")
+    public CategoryResponseDto updateCategoryById(
+            @PathVariable Long id,
+            @RequestBody @Valid CategoryRequestDto requestDto
+    ) {
+        return categoryService.updateCategoryById(id, requestDto);
+    }
+
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @DeleteMapping("/{id}")
```
Comment (d1sam) [inline src/main/java/mate/academy/bookstore/controller/CategoryController.java:70]:
```suggestion
    @ResponseStatus(HttpStatus.NO_CONTENT)
    @DeleteMapping("/{id}")
```

File: src/main/java/mate/academy/bookstore/model/Category.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id=?")
+@SQLRestriction("is_deleted=false")
+@Table(name = "categories")
+public class Category {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column(nullable = false)
+    private String name;
+    private String description;
```
Comment (d1sam) [inline src/main/java/mate/academy/bookstore/model/Category.java:27]:
```suggestion
    @Column
    private String description;
```

File: src/main/java/mate/academy/bookstore/model/Category.java [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import lombok.Data;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
+
+@Entity
+@Data
```
Comment (d1sam) [inline src/main/java/mate/academy/bookstore/model/Category.java:15]:
```suggestion
@Data
@NoArgsConstructor
```

**Review by d1sam — CHANGES_REQUESTED** [review]
good job, but some changes should be done!

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**